alibi/explainers/anchors/anchor_base.py

```diff
@@ -790,8 +790,11 @@ def anchor_beam(self, delta: float = 0.05, epsilon: float = 0.1, desired_confide
         # if no result is found, choose highest precision of best result candidate from every round
         if not best_anchor:
             success = False  # indicates the method has not found an anchor
-            logger.warning('Could not find an result satisfying the {} precision constraint. Now returning '
-                           'the best non-eligible result.'.format(desired_confidence))
+            logger.warning(f'Could not find an anchor satisfying the {desired_confidence} precision constraint. '
+                           f'Now returning the best non-eligible result. The desired precision threshold might not be '
+                           f'achieved due to the quantile-based discretisation of the numerical features since the '
+                           f'synthetic instances satisfying the anchor are constructed by sampling the numerical '
+                           f'features from their corresponding, potentially large, quantile intervals.')
```

Comment:
I'm sort of tempted to suggest just saying something like: "The resolution of bins may be too large to ensure we find an anchor of required precision. Note that higher resolution may not be possible due to sparsity of samples.". But I'm not sure... I think the above is fine as well.